javascript/javascript2/week3/fastestPresser/main.js

```diff
@@ -0,0 +1,101 @@
+// 1. Create an input and a button in html. When the button is clicked, get the value of the input.
+//This value will be the amount of time the game should run.
+// 2. Set a timeout for the time specified by the user.
+//After that time has run out just log out a simple string.
+// 3. Create an event listener so you can call a function when any key is pressed.
+//Now grap the actual key that was pressed.Fx was it a j or an i.We are interested in s and l.
+//Here google is your friend!
+// 4. Keep a counter for how many times l and s was pressed.
+// 5. Now put it all together! After the timeout is done figure out which of the counters is largest.
+const topContainerRef = document.getElementById("topContainer");
+const inputDurationRef = document.getElementById("inputDuration");
+const startBtnRef = document.getElementById("startBtn");
+const leftPlayerCounterOutputRef = document.getElementById(
+  "leftPlayerCounterOutput"
+);
+const rightPlayerCounterOutputRef = document.getElementById(
+  "rightPlayerCounterOutput"
+);
+const leftPlayerInterfaceRef = document.getElementById("leftPlayerInterface");
+const rightPlayerInterfaceRef = document.getElementById("rightPlayerInterface");
+const winNotificationRef = document.createElement("p");
+const confettiCanvas = document.createElement("canvas");
+
+let gameDuration = 0;
+let leftPlayerCounter = 0;
+let rightPlayerCounter = 0;
+
+inputDurationRef.addEventListener("input", onInputChange);
+startBtnRef.addEventListener("click", onStartBtnClick);
+
+function onInputChange(e) {
+  gameDuration = parseInt(e.target.value);
+  if (!isNaN(gameDuration)) {
+    startBtnRef.disabled = false;
+    return gameDuration;
+  }
+}
+
+function onKeyUpHandler(e) {
+  if (e.key === "s") {
+    leftPlayerCounter += 1;
+    leftPlayerCounterOutputRef.textContent = leftPlayerCounter;
+    return leftPlayerCounter;
+  } else if (e.key === "l") {
+    rightPlayerCounter += 1;
+    rightPlayerCounterOutputRef.textContent = rightPlayerCounter;
+    return rightPlayerCounter;
+  }
+}
+function onStartBtnClick(e) {
+  const duration = gameDuration * 1000;
+  setTimeout(endGame, duration);
+  document.addEventListener("keyup", onKeyUpHandler);
+}
+
+function getConfetti(winnerSide) {
+  confettiCanvas.setAttribute("id", "winner-canvas");
+  winnerSide.appendChild(confettiCanvas);
```

Comment:
I think in this case I would just put two canvas holders in the DOM, one for each player and just toggle the visibility before calling `confetti.render`. 
I think a) the code would be easier to understand and b) your performance would be slightly better. Having 1 extra DOM element is cheaper than adding and removing elements.